Source/pack.cpp

```diff
@@ -88,10 +88,12 @@ bool IsCreationFlagComboValid(uint16_t iCreateInfo)
 	const bool isUsefulItem = (iCreateInfo & CF_USEFUL) == CF_USEFUL;
 
 	if (isPregenItem)
+		// Pregen flags are discarded when an item is picked up, therefore impossible to have in the inventory
 		return false;

```

Comment:
I find it unusual to put comments inside single-line if statements in C++.
```suggestion
	if (isPregenItem) {
		// Pregen flags are discarded when an item is picked up, therefore impossible to have in the inventory
		return false;
	}
```